src/utils/hdKey.ts

```diff
@@ -95,17 +100,30 @@ const mnemonicToSeed = (mnemonic: string): Uint8Array => {
  * @param offset the offset used for key derivation, defaults to [HARDENED_OFFSET]
  */
 export const derivePrivateKeyFromMnemonic = (
-  keyType: KeyType,
+  scheme: SigningSchemeInput,
   path: string,
   seedPhrase: string,
   offset = HARDENED_OFFSET,
 ): DerivedKeys => {
-  if (!isValidPath(path)) {
-    throw new Error("Invalid derivation path");
+  // if (!isValidPath(path)) {
+  //   throw new Error("Invalid derivation path");
+  // }
+  let keyType;
+  switch (scheme) {
+    case SigningSchemeInput.Ed25519:
+      keyType = KeyType.ED25519;
+      break;
+    case SigningSchemeInput.Secp256k1Ecdsa:
+      keyType = KeyType.Secp256k1;
+      break;
+    default:
+      throw new Error(`Signing scheme ${scheme} not supported`);
   }
-
   // Derive the master key from the mnemonic
-  const { key, chainCode } = deriveKey(keyType, mnemonicToSeed(seedPhrase));
+  const { key, chainCode } = deriveKey(keyType, seedPhrase);
+  console.log("key", key);
+  console.log("chainCode", Hex.fromHexInput(chainCode).toString());
+  console.log("--------");
   const segments = splitPath(path).map((el) => parseInt(el, 10));
```

Comment:
@0xmaayan can you print the `key` and `chainCode` returned here? Rather than the ones above?